frontend/pkg/frontend/helpers.go

```diff
@@ -17,6 +18,57 @@ import (
 	"github.com/Azure/ARO-HCP/internal/database"
 )
 
+// CheckForConflict returns a "409 Conflict" error response if the provisioning
+// state of the resource is non-terminal, or any of its parent resources within
+// the same provider namespace are in a "Deleting" state.
+func (f *Frontend) CheckForConflict(ctx context.Context, operationRequest database.OperationRequest, doc *database.ResourceDocument) *arm.CloudError {
```

Comment:
idea: would this be clearer named as `CheckForProvisioningStateOperationConflict`? It is a longer name, but it might save us from having to write a clarifying comment on each call?

Reply:
I'm not opposed to a longer name, but the clarifying comment on each call is there to address error handling since there's a conspicuous (but intentional) absence of a log entry call in the error block.

```
	// CheckForConflict does not log conflict errors but
	// does log unexpected errors like database failures.
	cloudError := f.CheckForConflict(ctx, operationRequest, doc)
	if cloudError != nil {
		arm.WriteCloudError(writer, cloudError)  <-- usually preceded by f.logger.Error()
		return
	}
```

Reply:
I renamed it `CheckForProvisioningStateConflict`.